DQM/TrackingMonitorSource/plugins/StandaloneTrackMonitor.cc

```diff
@@ -258,222 +488,687 @@ StandaloneTrackMonitor::StandaloneTrackMonitor(const edm::ParameterSet& ps)
-    hOnTrkClusChargeThinH_ =
-        iBook.book1D("hOnTrkClusChargeThin", "On-track Cluster Charge (Thin Sensor)", 100, 0, 1000);
-  if (!hOnTrkClusWidthThinH_)
-    hOnTrkClusWidthThinH_ = iBook.book1D("hOnTrkClusWidthThin", "On-track Cluster Width (Thin Sensor)", 20, -0.5, 19.5);
-  if (!hOnTrkClusChargeThickH_)
-    hOnTrkClusChargeThickH_ =
-        iBook.book1D("hOnTrkClusChargeThick", "On-track Cluster Charge (Thick Sensor)", 100, 0, 1000);
-  if (!hOnTrkClusWidthThickH_)
-    hOnTrkClusWidthThickH_ =
-        iBook.book1D("hOnTrkClusWidthThick", "On-track Cluster Width (Thick Sensor)", 20, -0.5, 19.5);
-
-  if (!hOffTrkClusChargeThinH_)
-    hOffTrkClusChargeThinH_ =
-        iBook.book1D("hOffTrkClusChargeThin", "Off-track Cluster Charge (Thin Sensor)", 100, 0, 1000);
-  if (!hOffTrkClusWidthThinH_)
-    hOffTrkClusWidthThinH_ =
-        iBook.book1D("hOffTrkClusWidthThin", "Off-track Cluster Width (Thin Sensor)", 20, -0.5, 19.5);
-  if (!hOffTrkClusChargeThickH_)
-    hOffTrkClusChargeThickH_ =
-        iBook.book1D("hOffTrkClusChargeThick", "Off-track Cluster Charge (Thick Sensor)", 100, 0, 1000);
-  if (!hOffTrkClusWidthThickH_)
-    hOffTrkClusWidthThickH_ =
-        iBook.book1D("hOffTrkClusWidthThick", "Off-track Cluster Width (Thick Sensor)", 20, -0.5, 19.5);
+  hOnTrkClusChargeThinH_ = ibook.book1D("hOnTrkClusChargeThin", "On-track Cluster Charge (Thin Sensor)", 100, 0, 1000);
+  hOnTrkClusWidthThinH_ = ibook.book1D("hOnTrkClusWidthThin", "On-track Cluster Width (Thin Sensor)", 20, -0.5, 19.5);
+  hOnTrkClusChargeThickH_ =
+      ibook.book1D("hOnTrkClusChargeThick", "On-track Cluster Charge (Thick Sensor)", 100, 0, 1000);
+  hOnTrkClusWidthThickH_ =
+      ibook.book1D("hOnTrkClusWidthThick", "On-track Cluster Width (Thick Sensor)", 20, -0.5, 19.5);
+
+  hOffTrkClusChargeThinH_ =
+      ibook.book1D("hOffTrkClusChargeThin", "Off-track Cluster Charge (Thin Sensor)", 100, 0, 1000);
+  hOffTrkClusWidthThinH_ =
+      ibook.book1D("hOffTrkClusWidthThin", "Off-track Cluster Width (Thin Sensor)", 20, -0.5, 19.5);
+  hOffTrkClusChargeThickH_ =
+      ibook.book1D("hOffTrkClusChargeThick", "Off-track Cluster Charge (Thick Sensor)", 100, 0, 1000);
+  hOffTrkClusWidthThickH_ =
+      ibook.book1D("hOffTrkClusWidthThick", "Off-track Cluster Width (Thick Sensor)", 20, -0.5, 19.5);
 }
 void StandaloneTrackMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup) {
-  // Get event setup (to get global transformation)
+  if (verbose_)
+    std::cout << "Begin StandaloneTrackMonitor" << std::endl;
+
+  nevt++;
 
-  siStripClusterInfo_.initEvent(iSetup);
+  // Get event setup (to get global transformation)
+  const TrackerGeometry& tkGeom = (*tkGeom_);
 
   // Primary vertex collection
   edm::Handle<reco::VertexCollection> vertexColl;
   iEvent.getByToken(vertexToken_, vertexColl);
+  if (!vertexColl.isValid()) {
+    std::cerr << "Error! Failed to get reco::Vertex Collection, for " << vertexTag_ << std::endl;
```

Comment:
`cerr` is not thread safe. Please use either LogInfo or some other severity from the MessageLogging system.